hibernate-core/src/test/java/org/hibernate/test/annotations/onetoone/EmbeddedIdTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.test.annotations.onetoone;
+
+import java.io.Serializable;
+
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.EmbeddedId;
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+import org.junit.Test;
+
+@TestForIssue(jiraKey = "HHH-15235")
+public class EmbeddedIdTest extends BaseCoreFunctionalTestCase {
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class[] { Bar.class, Foo.class };
+	}
+
+	@Test
+	public void testMerge() {
+		inTransaction(
+				session -> {
+					FooId fooId = new FooId();
+					fooId.id = "foo";
+					Foo foo = new Foo();
+					foo.id = fooId;
+					Bar bar = new Bar();
+					BarId barId = new BarId();
+					barId.id = 1l;
+					bar.id = barId;
+					foo.bar = bar;
+					bar.foo = foo;
+					session.merge( foo );
+					session.flush();
```

Comment:
is this `flush()` important?
It's a bit puzzling; if it's important please comment about it, otherwise let's remove it.